js/completionCalculations.js

```diff
@@ -5,144 +5,152 @@ define([
-                "nonAssessmentTotal": nonAssessmentComponentsTotal,
-                "assessmentCompleted": assessmentComponentsCompleted,
-                "assessmentTotal": assessmentComponentsTotal
-            };
-
-            var showPageCompletionCourse = Adapt.course.get("_pageLevelProgress") && Adapt.course.get("_pageLevelProgress")._showPageCompletion !== false;
-            var showPageCompletionPage = contentObjectModel.get("_pageLevelProgress") && contentObjectModel.get("_pageLevelProgress")._showPageCompletion !== false;
-
-            if (showPageCompletionCourse && showPageCompletionPage) {
-                //optionally add one point extra for page completion to eliminate incomplete pages and full progress bars
-                // if _showPageCompletion is true then the progress bar should also consider it so add 1 to nonAssessmentTotal
-                pageCompletion.nonAssessmentCompleted += isComplete;
-                pageCompletion.nonAssessmentTotal += 1;
-            }
-
-            return pageCompletion;
-        }
-        // If it's a sub-menu
-        else if (viewType == 'menu') {
-
-            _.each(contentObjectModel.get('_children').models, function(contentObject) {
-                var completionObject = calculateCompletion(contentObject);
-                subProgressCompleted += contentObjectModel.subProgressCompleted || 0;
-                subProgressTotal += contentObjectModel.subProgressTotal || 0;
-                nonAssessmentComponentsTotal += completionObject.nonAssessmentTotal;
-                nonAssessmentComponentsCompleted += completionObject.nonAssessmentCompleted;
-                assessmentComponentsTotal += completionObject.assessmentTotal;
-                assessmentComponentsCompleted += completionObject.assessmentCompleted;
-            });
-
-            return {
-                "subProgressCompleted": subProgressCompleted,
-                "subProgressTotal" : subProgressTotal,
-                "nonAssessmentCompleted": nonAssessmentComponentsCompleted,
-                "nonAssessmentTotal": nonAssessmentComponentsTotal,
-                "assessmentCompleted": assessmentComponentsCompleted,
-                "assessmentTotal": assessmentComponentsTotal,
-            };
+        var viewType = contentObjectModel.get('_type');
+        var nonAssessmentComponentsTotal = 0;
+        var nonAssessmentComponentsCompleted = 0;
+        var assessmentComponentsTotal = 0;
+        var assessmentComponentsCompleted = 0;
+        var subProgressCompleted = 0;
+        var subProgressTotal = 0;
+        var isComplete = contentObjectModel.get("_isComplete") ? 1 : 0;
+        var children;
+
+        switch (viewType) {
+            case 'page':
+                // If it's a page
+                var descendantComponents = contentObjectModel.findDescendantModels('components');
+                children = descendantComponents.filter(function(comp) {
```

Comment:
could be switched to `children = contentObjectModel.findDescendantModels('components', { where: { _isAvailable: true, _isOptional: false }});`